catalog/core/catalog-core-standardframework/src/test/java/ddf/catalog/cache/solr/impl/CachingFederationStrategyTest.java

```diff
@@ -0,0 +1,748 @@
+        when(originalResults.take()).thenReturn(mockResult);
+
+        doAnswer(invocation -> results.add(mockResult)).when(offsetResultQueue)
+                .addResult(any(Result.class), any(Boolean.class));
+
+        offsetResultHandler.run();
+
+        assertThat(results.size(), is(1));
+        assertThat(results.get(0)
+                .getMetacard()
+                .getId(), is("mock metacard"));
+        verify(offsetResultQueue, atLeastOnce()).addResult(any(Result.class), any(Boolean.class));
+    }
+
+    @Test
+    public void testCatchPluginExecutionException() throws Exception {
+        PreFederatedQueryPlugin mockPlug = mock(PreFederatedQueryPlugin.class);
+        when(mockPlug.process(any(Source.class),
+                any(QueryRequest.class))).thenThrow(new PluginExecutionException());
+        strategy = new CachingFederationStrategy(queryExecutor,
+                Arrays.asList(mockPlug),
+                new ArrayList<>(),
+                cache,
+                cacheExecutor,
+                mock(ValidationQueryFactory.class),
+                new CacheQueryFactory(new GeotoolsFilterBuilder()));
+
+        QueryRequest fedQueryRequest = new QueryRequestImpl(mockQuery, properties);
+
+        strategy.federate(Arrays.asList(mock(Source.class)), fedQueryRequest);
+        verify(mockPlug).process(any(Source.class), any(QueryRequest.class));
+    }
+
+    @Test
+    public void testCatchStopProcessingException() throws Exception {
+        PreFederatedQueryPlugin mockPlug = mock(PreFederatedQueryPlugin.class), mockPlug2 = mock(
+                PreFederatedQueryPlugin.class);
+        when(mockPlug.process(any(Source.class),
+                any(QueryRequest.class))).thenThrow(new StopProcessingException("test exception"));
+        strategy = new CachingFederationStrategy(queryExecutor,
+                Arrays.asList(mockPlug, mockPlug2),
+                new ArrayList<>(),
+                cache,
+                cacheExecutor,
+                mock(ValidationQueryFactory.class),
+                new CacheQueryFactory(new GeotoolsFilterBuilder()));
+
+        QueryRequest fedQueryRequest = new QueryRequestImpl(mockQuery, properties);
+
+        strategy.federate(Arrays.asList(mock(Source.class)), fedQueryRequest);
+        // First plugin throws exception, so second plugin is untouched
+        verify(mockPlug).process(any(Source.class), any(QueryRequest.class));
+        verifyZeroInteractions(mockPlug2);
+    }
+
+    @Test
+    public void testCatalogProviderSource() throws Exception {
+        CatalogProvider catalogProviderSource = mock(CatalogProvider.class);
+        when(catalogProviderSource.getId()).thenReturn(SystemInfo.getSiteName());
+
+        QueryRequest fedQueryRequest = new QueryRequestImpl(mockQuery, properties);
+
+        strategy.federate(Arrays.asList(catalogProviderSource), fedQueryRequest);
+
+        verify(catalogProviderSource, atLeastOnce()).getId();
```

Comment:
❗️ We should verify that `validationQueryFactory. getQueryRequestWithValidationFilter()` has been called with the proper arguments.